app/mailers/user_mailer.rb

```diff
@@ -46,17 +44,82 @@ def work_collaborator(user, work)
     mail to: @user.email, subject: "You've been added to #{@work.title}"
   end
 
-  def nightly_user_activity(user)
-    @user = user
-    user_activity(@user)
-    if @active_user
-      mail to: @user.email, subject: "New FromThePage Activity"
+  def nightly_user_activity(user_activity)
+    @user_activity = user_activity
+
+    if @user_activity.has_contributions?
+      mail to: @user_activity.user.email, subject: "New FromThePage Activity"
     end
```

Comment:
All of the instance variables that were generated here but were hidden by a helper method have been encapsulated in a single `@user_activity` instance variable.